docs/extensibility/inputs.md

```diff
@@ -0,0 +1,151 @@
+# Arguments and flags
+
+Arguments and flags are the only way to get inputs from the end user and pass them to the config under the `with` field.
+
+Example flags and args:
+
+```yaml
+...
+args:
```

Comment:
How many arguments can we have?

Reply:
One, for now. In the future, we could support more args, but still have the `args` structure instead of an array.